### ISSUE TITLE
Check for unnecessary .name in `prefer-describe-function-title`

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [vitest-dev/eslint-plugin-vitest#743]
Suppose a test file imports a function in a way that's enforced by [`vitest/prefer-describe-function-title`](https://github.com/vitest-dev/eslint-plugin-vitest/blob/d48741a68294d109b263792ccbb7d6ba9c759e0f/docs/rules/prefer-describe-function-title.md). But instead of `describe(greet, ...`, the test actually uses `describe(greet.name, ...`:

```js
// greet.test.js
import { describe, expect, it } from "vitest";
import { greet } from "./greet.js";

describe(greet.name, () => {
    it("should return a greeting message", () => {
        expect(greet()).toBe("Hello, world!");
    });
});
```

```js
// greet.js
export function greet() {
    return "Hello, world!"
}
```

There's no need to refer to the function's `.name`. If the `greet` identifier is known to be a function with the same name text as the identifier then the `.name` is redundant. How about making `vitest/prefer-describe-function-title` consider a `.name` to be an equivalent string in that case?

```plaintext

describe(greet.name, () => {
//            ~~~~~
// Accessing the .name of this function is unnecessary.
```

It could come with an auto-fixer to remove the `.name`.

I'm not sure if this would be best as added behavior for the existing rule, or as a new rule altogether... but since this is so closely related to the existing rule, I'd lean towards the former personally.
